chat/adapter.py

```diff
@@ -0,0 +1,124 @@
+import json
+import sys
+import time
+import warnings
+from pathlib import Path
+from typing import Literal
+
+import lightning as L
+import torch
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from generate.base import generate
```

Comment:
You are using the generation code from the generation/base.py script, instead of chat/base.py.

Is that intended?

Reply:
When I use `from chat.base import generate`

There will be an issue on:

File "<path>/lit-gpt/lit_parrot/tokenizer.py", line 50, in decode
    tokens = [tensor.item()] if tensor.ndim == 0 else tensor.tolist()
AttributeError: 'generator' object has no attribute 'ndim'

But `from generate.base import generate` will be good.

Reply:
I see. Your current implementation won't work. chat.base.generate returns a generator of tokens whereas your current code expects a tensor of tokens. Please try out the `chat/base.py` script to understand how it works.

If the adapter checkpoints will always start with `### Response:`, we can simply ignore the tokens associated to that prefix.